src/Symfony/Bundle/SecurityBundle/Command/UserPasswordEncoderCommand.php

```diff
@@ -32,10 +35,10 @@ protected function configure()
     {
         $this
             ->setName('security:encode-password')
-            ->setDescription('Encode a password.')
+            ->setDescription('Encodes a password.')
             ->addArgument('password', InputArgument::OPTIONAL, 'Enter a password')
-            ->addArgument('user-class', InputArgument::OPTIONAL, 'Enter the user class configured to find the encoder you need.')
-            ->addArgument('salt', InputArgument::OPTIONAL, 'Enter the salt you want to use to encode your password.')
+            ->addArgument('user-class', InputArgument::OPTIONAL, 'Enter the user class configured to find the encoder you need.', 'Symfony\Component\Security\Core\User\User')
+            ->addOption('empty-salt', null, InputOption::VALUE_NONE, 'Do not generate a salt or let the encoder generate one.')
```

Comment:
@javiereguiluz I agree very strongly with the first two bullets, but not with the last one. I really don't see the value of being able to specify a salt, so even giving the option makes the command more complicated and the behaviour more unexpected (why accept a salt option and then discard it if encoding with bcrypt? You can totally do it, but it's almost always a really crappy idea - and that's equally true for *all* salt-using encoders, see what I mean? The only difference is the bcrypt encoder is smart enough to make using a cryptographically strong generated salt the default behaviour).

I think good DX is about making the right thing easy, and making mistakes more difficult; specifying a salt is almost always a mistake. For those instances where it's *not* a mistake, I think that's a sufficiently power-user situation that this command has no obligation to cater to it :stuck_out_tongue:

So IMO `--salt` should be removed. Ideally `--empty-salt` should also be removed *eventually*, but is probably necessary for some custom encoders until a PR fixes the leaky abstraction in the password encoder design.